Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Upsert.ts

```diff
@@ -113,9 +114,27 @@ export async function upsertDocument(
       // Perform the document upsert
       Logger.debug(`${moduleName}.upsertDocument Upserting document id ${id}`, traceId);
 
-      const { acknowledged, upsertedCount } = await mongoCollection.replaceOne({ _id: id }, document, asUpsert(session));
-      if (acknowledged) {
-        upsertResult.response = upsertedCount === 0 ? 'UPDATE_SUCCESS' : 'INSERT_SUCCESS';
+      const numberOfRetries: number = Config.get('MAX_NUMBER_OF_RETRIES') ?? 1;
+      const mongoUpsertResult = {
+        acknowledged: false,
+        upsertedCount: 0,
+      };
+
+      await retry(
+        async () => {
```

Comment:
I'm thinking you should instead abort and retry the whole transaction when any mongo call in the transaction throws an exception.

Separately, you'll want to sync with main, as RND-442 made a lot of changes, one of which is to extract much of the transaction block of upsert and update.

Reply:
Ok, sounds good.
Yes. I am working on the rebase. I'll probably end up closing this pull request and opening a new one.